packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx

```diff
@@ -30,8 +30,24 @@ interface FetchingWrappedAssetCardProps {
   sdk: FieldExtensionSDK;
 }
 
-function downloadAsset(url: string) {
-  window.open(url, '_blank', 'noopener,noreferrer');
+function downloadAsset(url: string, fileName: string) {
+  fetch(url, {
```

Comment:
I think this won't work when the target has [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) disabled, right?